### ISSUE TITLE
Question: How about introducing the `code-server` on the local development?

### ISSUE THREAD

**Issue by nyannyacha** [supabase/cli#1822]
I've seen a lot of posts about how the deno extension in vscode doesn't work well with existing project structures, making development difficult.

I think the main reason behind this is that it often conflicts with existing TypeScript projects, and I think it would be better to integrate `code-server` into the container image and isolate it from the existing development environment.

I integrated the `code-server` into the `Dockerfile`, and I think it's pretty good with development.

What do you guys think?

## Demo

https://github.com/supabase/edge-runtime/assets/46702285/a0f10c09-ed51-4217-9403-c8a998120bf7

😸

**Comment by laktek** [comment]
Done. transferred to supabase/cli

**Comment by silentworks** [comment]
I would vote for a no on this one as I wouldn't want the extra image/container just to have VS Code again as part of the CLI. This wouldn't be good for CI/CD as they would end up download this by default unless you use the exclude flag.

**Comment by nyannyacha** [comment]
> I would vote for a no on this one as I wouldn't want the extra image/container just to have VS Code again as part of the CLI. This wouldn't be good for CI/CD as they would end up download this by default unless you use the exclude flag.

I agree with you that `code-server` should not be added simply as an extra container or image to the cli.

The original idea was to introduce an extra build flag in the `edge-runtime` image to download and run the `code-server` binaries, so that they are only used for local development.

This way, there is no need to think about the extra overhead of introducing and maintaining an extra image into the cli codebase and only introduce just one flag to switch between the versions that include `code-server` binary or not.

**Comment by sweatybridge** [comment]
> This wouldn't be good for CI/CD as they would end up download this by default unless you use the exclude flag.

We have improved cli to only download images that are actually used. For eg. the migra image will be downloaded on the first time db diff command is run.
